feat(notification): реализовать модуль и gateway уведомлений

i-bocharov opened the pull request:
Добавлен NotificationModule и NotificationsGateway

Порт берётся из WS_NOTIFICATIONS_PORT (по умолчанию 4000)

Интеграция с WsJwtGuard

Добавлен тип WsSocket для безопасной типизации

Установлены типы @types/socket.io

Готов к интеграции с RequestsService

viktor-bolotin committed:
Реализован swagger для users

viktor-bolotin committed:
Реализован swagger для users

i-bocharov committed:
feat(wsConfig): добавить конфигурацию WebSocket через registerAs

- Создан ws.config.ts с WS_CONFIG и портом WS_NOTIFICATIONS_PORT
- Добавлена переменная WS_NOTIFICATIONS_PORT=4000 в .env.example

i-bocharov committed:
feat(notification): реализовать модуль и gateway уведомлений

- Добавлен NotificationModule и NotificationsGateway
- Порт берётся из WS_NOTIFICATIONS_PORT (по умолчанию 4000)
- Интеграция с WsJwtGuard
- Добавлен тип WsSocket для безопасной типизации
- Установлены типы @types/socket.io
- Готов к интеграции с RequestsService

i-bocharov committed:
refactor(test): исправить создание тестовой директории для загрузки файлов

i-bocharov committed:
refactor(test): настроить .env.test.local и DB config для E2E-тестов

i-bocharov committed:
feat(test): добавить globalSetup для очистки тестовой БД

i-bocharov committed:
refactor(test): добавить globalSetup в jest-e2e.json

i-bocharov committed:
feat(test): настроить E2E-тесты с изолированной БД и сидами

RolandSallaz commented on src/notification/notification.module.ts:
```diff
@@ -0,0 +1,13 @@
+import { WsJwtGuard } from './guards/ws-jwt.guard';
+import { wsConfig } from '../config/ws.config';
+
+@Module({
```
Теперь этот модуль нужно импортировать в App.module.ts, иначе к сокету не получится подключиться

Также требует установить драйвер https://disk.yandex.ru/i/gptfKe6gfXUasg `npm install @nestjs/platform-socket.io`

i-bocharov committed:
feat(notifications): подключён WebSocket-шлюз для уведомлений

- установлен @nestjs/platform-socket.io
- добавлен NotificationModule в AppModule

i-bocharov commented:
установлен @nestjs/platform-socket.io

добавлен NotificationModule в AppModule
